src/Workspaces/Core/Portable/FindSymbols/FindReferences/BaseTypeFinder.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+
+namespace Microsoft.CodeAnalysis.FindSymbols.FindReferences
+{
+    internal static partial class BaseTypeFinder
+    {
+        public static async Task<ImmutableArray<SymbolAndProjectId>> FindBaseTypesAndInterfacesAsync(
+            INamedTypeSymbol type, Project project, CancellationToken cancellationToken)
+        {
+            var typesAndInterfaces = FindBaseTypesAndInterfaces(type);
+            return await ConvertToSymbolAndProjectIdsAsync(typesAndInterfaces.CastArray<ISymbol>(), project, cancellationToken).ConfigureAwait(false);
+        }
+
+        public static async Task<ImmutableArray<SymbolAndProjectId>> FindOverriddenAndImplementedMembersAsync(
+            ISymbol symbol, Project project, CancellationToken cancellationToken = default)
```

Comment:
are these copied from somewhere... they seem super similar to helpers we already have.

Reply:
no default cancellation for internal members.  we don't want people forgetting to pass this value along.

Reply:
[https://github.com/dotnet/roslyn/blob/c420bead8f60f6582903b8d8676227bbd178f67b/src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs#L34](https://github.com/dotnet/roslyn/blob/c420bead8f60f6582903b8d8676227bbd178f67b/src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs#L34)

---
In reply to: [283962446](https://github.com/dotnet/roslyn/pull/35704#discussion_r283962446) [](ancestors = 283962446)